lib/temporal/workflow/history/event_target.rb

```diff
@@ -42,7 +44,7 @@ def self.workflow
         end
 
         def self.from_event(event)
-          _, target_type = TARGET_TYPES.find { |type, _| event.type.start_with?(type) }
+          _, target_type = TARGET_TYPES.find_all { |type, _| event.type.start_with?(type) }.last
```

Comment:
Why not sort the keys by length at load time to avoid having match multiple times?

Reply:
Yep, that does make more sense